jwt/jwt.go

```diff
@@ -122,6 +122,16 @@ func (jwt Decoded) Verify() error {
 		return fmt.Errorf("JWT signature verification failed: %w", err)
 	}
 
+	// check to ensure that the did used to sign the JWT matches the issuer claim.
+	// the value of KID should always be ${did}#${verificationMethodID} (aka did url)
+	if jwt.Claims.Issuer != "" && !strings.HasPrefix(jwt.Header.KID, jwt.Claims.Issuer) {
```

Comment:
if the issuer is empty should we err?

Reply:
great point @decentralgabe will fix now

Reply:
done: https://github.com/TBD54566975/web5-go/commit/b1b213f9460bc6aa9c0ec808eb874318b2d64f33